scoring/converter.py

```diff
@@ -1,12 +1,130 @@
-from sr.comp.scorer import Converter as BaseConverter
+from __future__ import annotations
 
+from sr.comp.match_period import Match
 
-class Converter(BaseConverter):
+
+def render_int(value: int | None) -> int | None:
+    """
+    Process a maybe missing integer value towards a canonical display form.
+    """
+    if not value:
+        # Display zeros as empty inputs
+        return None
+    return value
+
+
+def parse_int(value: str | None) -> int:
+    """
+    Parse a maybe missing integer value towards an integer.
+    """
+    if value is None or value == '':
+        return 0
+    return int(value)
+
+
+class Converter:
     def form_team_to_score(self, form, zone_id):
-        raise NotImplementedError
+        """
+        Prepare the part of the score dict for the given zone from the form data.
+        """
+        return {
+            'zone': zone_id,
+            'disqualified':
+                form.get(f'disqualified_{zone_id}', None) is not None,
+            'present':
+                form.get(f'present_{zone_id}', None) is not None,
+            'left_scoring_zone':
+                form.get(f'left_scoring_zone_{zone_id}') is not None,
+            'robot_asteroids':
+                parse_int(form.get(f'robot_asteroids_{zone_id}')),
+        }
+
+    def form_to_score(self, match, form):
+        """
+        Prepare a score dict for the given match and form dict.
+
+        This method is used to convert the submitted information for storage as
+        YAML in the compstate.
+        """
+        zone_ids = range(len(match.teams))
+
+        teams = {}
+        for zone_id in zone_ids:
+            tla = form.get(f'tla_{zone_id}', None)
+            if tla:
+                teams[tla] = self.form_team_to_score(form, zone_id)
+
+        zones = list(zone_ids) + ['other']
+        arena = {}
+        for zone in zones:
+            arena[zone] = {
+                'egg_on_planet': form.get(f'egg_on_planet_{zone_id}') is not None,
```

Comment:
These should all be `zone`, not `zone_id`!

Reply:
https://github.com/srobo/sr2024-comp/pull/1/commits/b88a32b7b227521d131fc190c1335949331bbbf9